### DISCUSSION TITLE
分享脚本：使用GitHub Actions的Windows环境自动编译论文以获得正确字体

### DISCUSSION THREAD

**Discussion by zeyu42** [tuna/thuthesis#824] [Show and tell]
对于Mac、Linux和Overleaf用户，虽然可以用thuthesis编译论文，但字体并不严格符合学校的规定。我写了一个脚本，可以使用GitHub Actions自动编译论文。借助GitHub Actions提供的Windows环境，自动生成论文并使用正确的（Windows自带的）中文字体，欢迎使用：https://github.com/zeyu42/thuthesis-github-actions 😀

**Comment by Harry-Chen** [comment]
> 之所以要这样做而不是本地编译，是为了方便非 Windows 环境的用户在编译论文时使用正确的字体。实际上，学校提供了论文的 Word 模版，并在其中规定了字体。thuthesis 只有在 Windows 环境下编译才严格符合学校的规定。

这段描述不准确，事实上，无论在什么系统上，thuthesis 都会尝试选择 Windows 字体来编译。所以只要有这些字体（比如 macOS 安装了 Office，或者从 Windows 拷贝），就可以工作。

**Reply by zeyu42** [reply]
原来如此。不过我的Mac上安装了Office，thuthesis还是没法正常调用SimSun。不过还是感谢指出这个问题，我去修改一下我repo里的说明。

**Reply by Harry-Chen** [reply]
Office 的字体可能在自己的目录下，没有安装到系统全局，手工拷贝出来就行了
